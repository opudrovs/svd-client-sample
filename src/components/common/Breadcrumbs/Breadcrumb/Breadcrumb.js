import { memo } from 'react';

import PropTypes from 'prop-types';

const classNames = require('classnames');

import styles from './Breadcrumb.module.scss';

/**
 * Breadcrumb.
 */

const Breadcrumb = ({ text, url }) => (
    url
        ?
        <a href={url} className={classNames(styles.breadcrumb, styles.link)}>{text}</a>
        :
        <span className={styles.breadcrumb}>{text}</span>
);

Breadcrumb.propTypes = {
    text: PropTypes.string.isRequired,
    url: PropTypes.string
};

export default memo(Breadcrumb);
