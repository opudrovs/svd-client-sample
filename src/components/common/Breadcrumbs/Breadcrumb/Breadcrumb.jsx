import { memo } from 'react';

import PropTypes from 'prop-types';

import Link from 'next/link';

import classNames from 'classnames';

import styles from './Breadcrumb.module.scss';

/**
 * Breadcrumb component.
 */
const Breadcrumb = ({ text, href }) =>
  href ? (
    <Link href={href} className={classNames(styles.breadcrumb, styles.link)}>
      {text}
    </Link>
  ) : (
    <span className={styles.breadcrumb}>{text}</span>
  );

Breadcrumb.propTypes = {
  text: PropTypes.string.isRequired,
  href: PropTypes.string,
};

export default memo(Breadcrumb);
