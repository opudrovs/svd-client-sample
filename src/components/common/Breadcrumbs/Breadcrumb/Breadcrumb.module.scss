.breadcrumb {
    @include linkPadding;
}

.link {
    @include linkText;
}
