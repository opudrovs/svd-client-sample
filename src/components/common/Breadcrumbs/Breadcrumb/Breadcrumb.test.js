/* REACT */

import React from 'react';

/* LIBRARIES */

import { shallow } from 'enzyme';
import renderer from 'react-test-renderer';

/* COMPONENTS */

import Breadcrumb from './Breadcrumb.js';

const text = 'Test text';

const href = 'https://www.smartvectordesign.com';


/* TESTS */

describe('With Enzyme', () => {
    it(`Component shows '${text}'`, () => {
        const component = shallow(<Breadcrumb text={text} />);

        expect(component.find('span')
            .text())
            .toEqual(text);
    });
});

describe('With Enzyme', () => {
    it(`Component shows '${text}' and has '${href}' for href`, () => {
        const component = shallow(<Breadcrumb href={href} text={text} />);

        expect(component.find('a')
            .text())
            .toEqual(text);

        expect(component.find('Link').prop('href'))
            .toEqual(href);
    });
});

/* SNAPSHOT TESTS */

describe('With Snapshot Testing', () => {
    it(`Component shows '${text}'`, () => {
        const component = renderer.create(<Breadcrumb text={text} />);
        const tree = component.toJSON();
        expect(tree)
            .toMatchSnapshot();
    });
});

describe('With Snapshot Testing', () => {
    it(`Component shows '${text}' and has '${href}' for href`, () => {
        const component = renderer.create(<Breadcrumb href={href} text={text} />);
        const tree = component.toJSON();

        expect(tree)
            .toMatchSnapshot();
    });
});