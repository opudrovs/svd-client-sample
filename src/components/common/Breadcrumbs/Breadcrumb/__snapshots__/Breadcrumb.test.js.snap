// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Snapshot Testing Component shows 'Test text' 1`] = `
<span
  className="breadcrumb"
>
  Test text
</span>
`;

exports[`With Snapshot Testing Component shows 'Test text' and has 'https://www.smartvectordesign.com' for href 1`] = `
<a
  className="breadcrumb link"
  href="https://www.smartvectordesign.com"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Test text
</a>
`;
